analyzer-core/src/main/java/ffb/analyzer/core/queue/Task.java

```diff
@@ -0,0 +1,43 @@
+package ffb.analyzer.core.queue;
+
+import java.util.function.Consumer;
+
+public class Task<T> {
+    public int id;
+
+    private T objectToPerformTaskOn;
+    private Consumer<T> actionToPerform;
+
+
+    public void performTask() {
```

Comment:
Thoughts on having this method just throw the exception and having `PriorityQueueManger` handle logging?

Reply:
Makes sense, goes with the Manager structure better. Updated.